### ISSUE TITLE
About training details and performance

### ISSUE THREAD

**Issue by zhaoyang10** [MandyMo/pytorch_HMR#31]
Hello @MandyMo ,
    Thank you for releasing the code, it is a really great work.
    I trained with this code and got reasonable results on 3d datasets, but terrible results on 2d datasets. The losses all seem stable. Then I tried to change hyperparameters, but the results on 2d datasets were far from good. I don't know whether it's normal or not. Really confused.
    How is the performance of your pre-trained model?
    What's your performance on training set and testing set of 2d datasets?
    Can you share the hyperparameter setting or give some instructions?
    Thanks a lot! Following are my training details and results.

    This is the default parameters setting from this code, just changed the batch_size_(2d, 3d, adv) to fulfill my GPUs. I trained 7M samples for each dataset, you can regard it as about 1M iterations with batch size of 7.
    Losses:
![2019-06-29_102414](https://user-images.githubusercontent.com/10057534/60378809-e454af00-9a5a-11e9-9e0e-13ab7c0770ad.png)
    Training results from 3d datasets, they seem reasonable.
![epoch781_000_3d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378830-3b5a8400-9a5b-11e9-955d-cf98f2a5585f.jpg)
    Training results from 2d datasets, they seem terrible, but the loss for 2d points keeps stable. 
![epoch781_000_2d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378835-65ac4180-9a5b-11e9-8bd3-aa92fcc512a9.jpg)

    Then I amplified the ratio of 2d loss for 10 times, trying to lower 2d loss. But the result didn't goes the way I wanted. The 2d loss went a little bit lower if it divides 10 comparing the former one, it still much higher than 3d keypoint loss. And the results on 2d datasets were not satisfying.
    The losses:
![2019-06-29_104618](https://user-images.githubusercontent.com/10057534/60378871-f420c300-9a5b-11e9-80f0-a5732b1b82af.png)
 
     Training results from 3d datasets:
![epoch561_000_3d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378874-19adcc80-9a5c-11e9-977f-4ad079bb4912.jpg)
    Training results from 2d datasets:
![epoch561_000_2d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378880-2500f800-9a5c-11e9-9e83-e67ec498fbca.jpg)

**Comment by zhaoyang10** [comment]
@willie1997 @hsyntemiz @mehameha998
I read the histories of issues, how about your training results?
Thanks a lot.

**Comment by eng100200** [comment]
@zhaoyang10  what do u mean by 2d datasets?

**Comment by zhaoyang10** [comment]
It means datasets with only 2d lables, such as COCO, lsp and so on.

**Comment by eng100200** [comment]
@zhaoyang10  ok,,,,i think the model is for 2d input images ,, i did not read this work completely,,but, my understanding is this

**Comment by eng100200** [comment]
@zhaoyang10 do you use wechat?

**Comment by zhaoyang10** [comment]
You can send me e-mails. It's not convenient to release my wechat directly.

**Comment by eng100200** [comment]
@zhaoyang10  my email sm_Adnan21@hotmail.com

**Comment by eng100200** [comment]
@zhaoyang10  added

**Comment by junhaozhang98** [comment]
Hi,I've just started to learn human pose recognition. I'm very confused about dealing with the original data set. Could you please sent all annot.h5 data set and model.zip to my email 2868627077@qq.com? Thank you very much!!!!!!!!!

**Comment by junhaozhang98** [comment]
@zhaoyang10 Hi,I've just started to learn human pose recognition. I'm very confused about dealing with the original data set. Could you please sent all annot.h5 data set and model.zip to my email 2868627077@qq.com? Thank you very much!!!!!!!!!

**Comment by RuiboFan** [comment]
> Hello @MandyMo , Thank you for releasing the code, it is a really great work. I trained with this code and got reasonable results on 3d datasets, but terrible results on 2d datasets. The losses all seem stable. Then I tried to change hyperparameters, but the results on 2d datasets were far from good. I don't know whether it's normal or not. Really confused. How is the performance of your pre-trained model? What's your performance on training set and testing set of 2d datasets? Can you share the hyperparameter setting or give some instructions? Thanks a lot! Following are my training details and results.
> 
> ```
> This is the default parameters setting from this code, just changed the batch_size_(2d, 3d, adv) to fulfill my GPUs. I trained 7M samples for each dataset, you can regard it as about 1M iterations with batch size of 7.
> Losses:
> ```
> 
> ![2019-06-29_102414](https://user-images.githubusercontent.com/10057534/60378809-e454af00-9a5a-11e9-9e0e-13ab7c0770ad.png) Training results from 3d datasets, they seem reasonable. ![epoch781_000_3d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378830-3b5a8400-9a5b-11e9-955d-cf98f2a5585f.jpg) Training results from 2d datasets, they seem terrible, but the loss for 2d points keeps stable. ![epoch781_000_2d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378835-65ac4180-9a5b-11e9-8bd3-aa92fcc512a9.jpg)
> 
> ```
> Then I amplified the ratio of 2d loss for 10 times, trying to lower 2d loss. But the result didn't goes the way I wanted. The 2d loss went a little bit lower if it divides 10 comparing the former one, it still much higher than 3d keypoint loss. And the results on 2d datasets were not satisfying.
> The losses:
> ```
> 
> ![2019-06-29_104618](https://user-images.githubusercontent.com/10057534/60378871-f420c300-9a5b-11e9-80f0-a5732b1b82af.png)
> 
> ```
>  Training results from 3d datasets:
> ```
> 
> ![epoch561_000_3d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378874-19adcc80-9a5c-11e9-977f-4ad079bb4912.jpg) Training results from 2d datasets: ![epoch561_000_2d_img_rend_cv2](https://user-images.githubusercontent.com/10057534/60378880-2500f800-9a5c-11e9-9e83-e67ec498fbca.jpg)

do you have the training datasets?

**Comment by Ethan-cpp** [comment]
can I use your code ? Thank you! I can`t run the code @zhaoyang10

**Comment by Ethan-cpp** [comment]
could you help me run the code? I am terrible. Thank you very much!
